ethereumj-core/src/main/java/org/ethereum/listener/BackwardCompatibilityEthereumListenerProxy.java

```diff
@@ -0,0 +1,156 @@
+package org.ethereum.listener;
+
+import org.ethereum.core.*;
+import org.ethereum.net.eth.message.StatusMessage;
+import org.ethereum.net.message.Message;
+import org.ethereum.net.p2p.HelloMessage;
+import org.ethereum.net.rlpx.Node;
+import org.ethereum.net.server.Channel;
+import org.ethereum.publish.Publisher;
+import org.ethereum.publish.event.Events;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * Backward compatibility component that holds old and new event publishing implementations.<br>
+ * Proxies all events to both implementations.
+ * Will be removed after {@link EthereumListener} eviction.
+ *
+ * @author Eugene Shevchenko
+ */
+@Primary
+@Component
+public class BackwardCompatibilityEthereumListenerProxy implements EthereumListener {
+
+    private final CompositeEthereumListener compositeListener;
+    private final Publisher publisher;
+
+    @Autowired
+    public BackwardCompatibilityEthereumListenerProxy(CompositeEthereumListener listener, Publisher publisher) {
+        this.compositeListener = listener;
+        this.publisher = publisher;
+    }
+
+    public CompositeEthereumListener getCompositeListener() {
+        return compositeListener;
+    }
+
+    public Publisher getPublisher() {
+        return publisher;
+    }
+
+    public void addListener(EthereumListener listener) {
+        this.compositeListener.addListener(listener);
+    }
+
+    @Override
+    public void trace(String output) {
+        compositeListener.trace(output);
+    }
+
+    @Override
+    public void onNodeDiscovered(Node node) {
+        compositeListener.onNodeDiscovered(node);
+        publisher.publish(Events.onNodeDiscovered(node));
```

Comment:
Maybe use static import for `Events`?

Reply:
Nope, we have several methods with the same signature in both classes. That's why we should use full name in this case. In addition it's temporary class, so it's not critical I suppose.